controllers/etcdpeer_controller.go

```diff
@@ -16,14 +20,104 @@ type EtcdPeerReconciler struct {
 	Log logr.Logger
 }
 
+func ignoreNotFound(err error) error {
+	if apierrs.IsNotFound(err) {
+		return nil
+	}
+	return err
+}
+
+const (
+	EtcdImage = "quay.io/coreos/etcd:v3.2.27"
+)
+
 // +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdpeers,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdpeers/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=apps,resources=replicaset,verbs=get;update;patch;create
+
+func defineReplicaSet(etcdPeer etcdv1alpha1.EtcdPeer) (appsv1.ReplicaSet, error) {
+	var replicas int32 = 1
+
+	return appsv1.ReplicaSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:      make(map[string]string),
+			Annotations: make(map[string]string),
+			Name:        etcdPeer.Name,
+			Namespace:   etcdPeer.Namespace,
+		},
+		Spec: appsv1.ReplicaSetSpec{
+			// This will *always* be 1. Other peers are handled by other EtcdPeers.
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app":  "etcd",
+					"peer": etcdPeer.Name,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app":  "etcd",
+						"peer": etcdPeer.Name,
+					},
+					Annotations: make(map[string]string),
+					Name:        etcdPeer.Name,
+					Namespace:   etcdPeer.Namespace,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "etcd",
+							Image: EtcdImage,
+						},
+					},
+				},
+			},
+		},
+	}, nil
+}
 
 func (r *EtcdPeerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("etcdpeer", req.NamespacedName)
+	ctx := context.Background()
+	log := r.Log.WithValues("etcdpeer", req.NamespacedName)
+
+	var etcdPeer etcdv1alpha1.EtcdPeer
+	if err := r.Get(ctx, req.NamespacedName, &etcdPeer); err != nil {
+		log.Error(err, "unable to fetch EtcdPeer")
+		return ctrl.Result{}, ignoreNotFound(err)
+	}
+
+	log.V(1).Info("Found EtcdPeer",
```

Comment:
What is `V(1)` meant to denote? i.e. what log level? I think standard Info level is `0` but not certain 🤔

Reply:
(and if using the standard level, `V` isn't needed at all

Reply:
I was referring to the examples from the [Kubebuilder book](https://book.kubebuilder.io/cronjob-tutorial/controller-implementation.html) which used the `log.V(1)` form. `log` is actually shadowed here too, to be the result of `r.Log.WithValues("etcdpeer", req.NamespacedName)`.

I will simplify this though.

Reply:
So the `V(1)` is because this is a `"github.com/go-logr/logr"` logger. The "V-Level" [is the verbosity](https://github.com/go-logr/logr#why-v-levels).

So this probably should be a higher V-Level, maybe a 4 or so. I'll adjust that.